icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -4217,6 +4217,7 @@ SimpleDateFormat::tzFormat(UErrorCode &status) const {
 void SimpleDateFormat::parsePattern() {
     fHasMinute = FALSE;
     fHasSecond = FALSE;
+    fHasHanYearChar = FALSE;
```

Comment:
What is this line for?

Reply:
It looks like the branch doesn't have the fix a9b26a447d688884a4bf5ed091e46470ed250f70 from master.